sdk/identity/azure-identity/azure/identity/_credentials/user.py

```diff
@@ -160,7 +160,10 @@ def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
 
         token = self._client.obtain_token_by_refresh_token(scopes, self._username)
         if not token:
-            raise ClientAuthenticationError(message="No cached token found for '{}'".format(self._username))
+            if self._username:
+                raise ClientAuthenticationError(message="No cached token found for '{}'".format(self._username))
+            else:
```

Comment:
pylint complains about the unnecessary `else` (here and in the async equivalent). Several ways to satisfy it, what do you think of this one:
```py
message = "No refresh token found"
if self._username:
    message += " for user '{}'".format(self._username)
raise ClientAuthenticationError(message=message)
```

Reply:
SGTM